djinn/__init__.py

```diff
@@ -12,13 +15,23 @@ def __init__(self, jenkinsurl=None, dburl=None):
         """
         self.logger = get_named_logger('Djinn')
         self.dj = DJenkins(url=jenkinsurl, logger=self.logger)
+        self.dburl = dburl
         self.db = PipelineResults(connection_url=dburl, echo=False)
+        self.service = AnalysisService(analysis=Analysis(), pipeline=self.db)
 
     def get_all_pipeline_results_and_save_to_db(self, pipelinebranch):
         """
         Fetch all repository pipeline data and write to database.
+        Could be memory-hungry depending on your instance - may be better to break it down by folder if so.
```

Comment:
Jenkins? Not the way this method works - it's actually hitting the API separately for every repo and collating the results. Could be rewritten to perform writes to the db per repo by more or less copying the function this is calling.

For reference, it didn't even blink at our test instance - mem usage has been lower than 50MB constantly. If you had thousands of repos it'd be a problem, possibly.